File: lab2_task3/main.php
<?php

function countSequences($sequence) {
    if (count($sequence) < 2) return 0;
    
    $seqLen = 1;
    $seqCount = 0;
    
    for ($i = 0; $i < count($sequence) - 1; $i++) {
        if ($sequence[$i+1] >= $sequence[$i]) {
            $seqLen++;
        } elseif ($seqLen > 1) {
            $seqCount++;
            $seqLen = 1;
        } else {
            $seqLen = 1;
        }
    }
    
    if ($seqLen > 1) $seqCount++;
    
    return $seqCount;
}

echo "Введите последовательность чисел через пробел: ";
$input = trim(fgets(STDIN));
$nums = array_map('intval', explode(' ', $input));
echo countSequences($nums) . "\n";
?>

File: lab2_task3/main.rs
use std::io;

fn count_sequences(sequence: &[i32]) -> i32 {
    if sequence.len() < 2 {
        return 0;
    }

    let mut seq_len = 1;
    let mut seq_count = 0;

    for i in 0..sequence.len() - 1 {
        if sequence[i+1] >= sequence[i] {
            seq_len += 1;
        } else if seq_len > 1 {
            seq_count += 1;
            seq_len = 1;
        } else {
            seq_len = 1;
        }
    }

    if seq_len > 1 {
        seq_count += 1;
    }

    seq_count
}

fn main() {
    println!("Введите последовательность чисел через пробел:");
    let mut input = String::new();
    io::stdin().read_line(&mut input).unwrap();
    
    let nums: Vec<i32> = input
        .trim()
        .split_whitespace()
        .map(|s| s.parse().unwrap())
        .collect();
    
    println!("{}", count_sequences(&nums));
}